### PR TITLE
[DROOLS-7172] Drools 8 docs : Retired features and components

### PR DESCRIPTION
**Pull request by tkobayas**
**JIRA**: 
https://issues.redhat.com/browse/DROOLS-7172

<details>
<summary>
How to replicate CI configuration locally?
</summary>

Build Chain tool does "simple" maven build(s), the builds are just Maven commands, but because the repositories relates and depends on each other and any change in API or class method could affect several of those repositories there is a need to use [build-chain tool](https://github.com/kiegroup/github-action-build-chain) to handle cross repository builds and be sure that we always use latest version of the code for each repository.
 
[build-chain tool](https://github.com/kiegroup/github-action-build-chain) is a build tool which can be used on command line locally or in Github Actions workflow(s), in case you need to change multiple repositories and send multiple dependent pull requests related with a change you can easily reproduce the same build by executing it on Github hosted environment or locally in your development environment. See [local execution](https://github.com/kiegroup/github-action-build-chain#local-execution) details to get more information about it.
</details>

<details>
<summary>
How to retest this PR or trigger a specific build:
</summary>

- for <b>pull request checks</b>  
  Please add comment: <b>Jenkins retest this</b>

- for a <b>specific pull request check</b>  
  Please add comment: <b>Jenkins (re)run [drools|kogito-runtimes|kogito-apps|kogito-examples] tests</b>

- for a <b>full downstream build</b> 
  - for <b>jenkins</b> job: please add comment: <b>Jenkins run fdb</b>
  - for <b>github actions</b> job: add the label `run_fdb`

- <b>a compile downstream build</b> please  add comment: <b>Jenkins run cdb</b>

- <b>a full production downstream build</b> please add comment: <b>Jenkins execute product fdb</b>

- <b>an upstream build</b> please add comment: <b>Jenkins run upstream</b>

- for <b>quarkus branch checks</b>  
  Run checks against Quarkus current used branch  
  Please add comment: <b>Jenkins run quarkus-branch</b>

- for a <b>quarkus branch specific check</b>  
  Run checks against Quarkus current used branch  
  Please add comment: <b>Jenkins (re)run [drools|kogito-runtimes|kogito-apps|kogito-examples] quarkus-branch</b>

- for <b>quarkus main checks</b>  
  Run checks against Quarkus main branch  
  Please add comment: <b>Jenkins run quarkus-main</b>

- for a <b>specific quarkus main check</b>  
  Run checks against Quarkus main branch  
  Please add comment: <b>Jenkins (re)run [drools|kogito-runtimes|kogito-apps|kogito-examples] quarkus-main</b>

- for <b>quarkus lts checks</b>  
  Run checks against Quarkus lts branch  
  Please add comment: <b>Jenkins run quarkus-lts</b>

- for a <b>specific quarkus lts check</b>  
  Run checks against Quarkus lts branch  
  Please add comment: <b>Jenkins (re)run [drools|kogito-runtimes|kogito-apps|kogito-examples] quarkus-lts</b>

- for <b>native checks</b>  
  Run native checks  
  Please add comment: <b>Jenkins run native</b>

- for a <b>specific native check</b>  
  Run native checks 
  Please add comment: <b>Jenkins (re)run [drools|kogito-runtimes|kogito-apps|kogito-examples] native</b>

- for <b>mandrel checks</b>  
  Run native checks against Mandrel image
  Please add comment: <b>Jenkins run mandrel</b>

- for a <b>specific mandrel check</b>  
  Run native checks against Mandrel image  
  Please add comment: <b>Jenkins (re)run [drools|kogito-runtimes|kogito-apps|kogito-examples] mandrel</b>

 - for <b>mandrel lts checks</b>  
  Run native checks against Mandrel image and quarkus lts branch
  Please add comment: <b>Jenkins run mandrel-lts</b>

- for a <b>specific mandrel lts check</b>  
  Run native checks against Mandrel image and quarkus lts branch
  Please add comment: <b>Jenkins (re)run [drools|kogito-runtimes|kogito-apps|kogito-examples] mandrel-lts</b>
</details>

### CODE REVIEW

**Review by tarilabs — APPROVED** [review]
Thanks for these contributions, as suggested I would rename "removed" into "retired", otherwise is good with me! 👍

File: drools-docs/src/modules/ROOT/pages/migration-guide/_missing_features_components.adoc [hunk 1]
```diff
@@ -0,0 +1,20 @@
+[id='missing-features-components_{context}']
+
+[id='kie-server_{context}']
+== {KIE_SERVER}
+
+{KIE_SERVER} is retired and no longer a component of {PRODUCT} 8. If your system is working on {KIE_SERVER} with {PRODUCT} 7, consider migration to https://kogito.kie.org/[{KOGITO}].
+
+While {KIE_SERVER} hosts multiple kjar containers, One {KOGITO} instance hosts one domain service. Hence, you would create {KOGITO} project per kjar container, which would be microservice style.
+
+You can find detailed migration steps in this article https://blog.kie.org/2021/09/bringing-drools-rules-into-the-cloud-with-kogito-a-step-by-step-path.html[Bringing Drools rules into the cloud with Kogito: a step by step path]
```
Comment (tkobayas) [inline drools-docs/src/modules/ROOT/pages/migration-guide/_missing_features_components.adoc]: This part will likely be enhanced by @tarilabs 's https://issues.redhat.com/browse/DROOLS-7173 Thanks!

Comment (kaldesai) [inline drools-docs/src/modules/ROOT/pages/migration-guide/_missing_features_components.adoc]:
```suggestion
You can find the detailed migration steps in the https://blog.kie.org/2021/09/bringing-drools-rules-into-the-cloud-with-kogito-a-step-by-step-path.html[Bringing Drools rules into the cloud with Kogito: a step by step path] article.
```

File: drools-docs/src/modules/ROOT/pages/migration-guide/_missing_features_components.adoc [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+[id='business-central_{context}']
+== {BUSINESS_CENTRAL}
+
+{BUSINESS_CENTRAL} is retired and no longer a component of {PRODUCT} 8. For asset management, usual version control system is recommended, for example, git. For editors, https://marketplace.visualstudio.com/items?itemName=kie-group.vscode-extension-kogito-bundle[VS Code Kogito extension] is recommended.
```
Comment (kaldesai) [inline drools-docs/src/modules/ROOT/pages/migration-guide/_missing_features_components.adoc]:
```suggestion
{BUSINESS_CENTRAL} is retired and no longer a component of {PRODUCT} 8. For asset management, the usual version control system is recommended, for example, git. For editors, https://marketplace.visualstudio.com/items?itemName=kie-group.vscode-extension-kogito-bundle[VS Code Kogito extension] is recommended.
```

File: drools-docs/src/modules/ROOT/pages/release-notes/index.adoc [hunk 3]
```diff
@@ -39,6 +39,12 @@ You can refer the xref:KIE/index.adoc#_securitymanager[Security Manager chapter]
 
 The `KieMarshallers` are still present and working as expected on the release of the Drools 8-series, however we are currently exploring alternative solutions more idiomatic to cloud-native use cases.
 
+=== {KIE_SERVER} is retired
+{KIE_SERVER} is no longer a component of {PRODUCT} 8. See xref:migration-guide/index.adoc#kie-server_migration-guide[Migration guide chapter] for migration.
```
Comment (kaldesai) [inline drools-docs/src/modules/ROOT/pages/release-notes/index.adoc]:
```suggestion
{KIE_SERVER} is no longer a component of {PRODUCT} 8.  For migration, see xref:migration-guide/index.adoc#kie-server_migration-guide[Migration guide chapter].
```

File: drools-docs/src/modules/ROOT/pages/release-notes/index.adoc [hunk 4]
```diff
@@ -39,6 +39,12 @@ You can refer the xref:KIE/index.adoc#_securitymanager[Security Manager chapter]
...
+
+=== {BUSINESS_CENTRAL} is retired
+{BUSINESS_CENTRAL} is no longer a component of {PRODUCT} 8. See xref:migration-guide/index.adoc#business-central_migration-guide[Migration guide chapter] for migration.
```
Comment (kaldesai) [inline drools-docs/src/modules/ROOT/pages/release-notes/index.adoc]:
```suggestion
{BUSINESS_CENTRAL} is no longer a component of {PRODUCT} 8.  For migration, see xref:migration-guide/index.adoc#business-central_migration-guide[Migration guide chapter].
```

**Review by kaldesai — APPROVED** [review]
@tkobayas, Very minor suggestions. Otherwise, your content looks good to me. Please reach out if you have any doubts regarding the suggestions. Thank you!

**Comment by tkobayas** [comment]
Thank you, @kaldesai !

**Comment by tkobayas** [comment]
@tarilabs Please merge this PR when CI is done, thanks! (GHA is pending at the moment. I think it's safe to merge, though)

**Review by mariofusco — APPROVED** [review]

File: drools-docs/src/modules/ROOT/pages/migration-guide/_missing_features_components.adoc (hunk 1 above)
Comment (tarilabs) [reply drools-docs/src/modules/ROOT/pages/migration-guide/_missing_features_components.adoc]: ref https://github.com/kiegroup/drools/pull/4706#pullrequestreview-1117064506

blog post link moved to DRL section for preservation too

**Pull request merged**